app/src/test/java/org/mozilla/fenix/perf/PerformanceInflaterTest.kt

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.perf
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.FrameLayout
+import mozilla.components.support.test.robolectric.testContext
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.FenixRobolectricTestRunner
+import java.io.File
+
+@RunWith(FenixRobolectricTestRunner::class)
+class PerformanceInflaterTest {
+
+    private lateinit var perfInflater: MockInflater
+
+    private val layoutNotToTest = setOf(
+        "fragment_browser",
+        "fragment_add_on_internal_settings"
+    )
+
+    @Before
+    fun setup() {
+        InflationCounter.inflationCount.set(0)
+
+        perfInflater = MockInflater(LayoutInflater.from(testContext), testContext)
+    }
+
+    @Test
+    fun `WHEN we inflate a view,THEN the inflation counter should increase`() {
+        assertEquals(0, InflationCounter.inflationCount.get())
+        perfInflater.inflate(R.layout.fragment_home, null, false)
+        assertEquals(1, InflationCounter.inflationCount.get())
+    }
+
+    @Test
+    fun `WHEN inflating one of our resource file, the inflater should not crash`() {
+        val fileList = File("./src/main/res/layout").listFiles()
+        if (fileList != null) {
```

Comment:
When would it be null? I assume never – shouldn't we assert the list isn't null and has entries instead of silently passing the test if the list is null?